ocean_lib/models/v4/test/test_side_staking.py

```diff
@@ -0,0 +1,467 @@
+#
+# Copyright 2021 Ocean Protocol Foundation
+# SPDX-License-Identifier: Apache-2.0
+#
+import pytest
+from web3 import exceptions
+
+from ocean_lib.models.v4.bpool import BPool
+from ocean_lib.models.v4.erc20_token import ERC20Token
+from ocean_lib.models.v4.erc721_factory import ERC721FactoryContract
+from ocean_lib.models.v4.models_structures import PoolData
+from tests.resources.helper_functions import (
+    get_address_of_type,
+    deploy_erc721_erc20,
+)
+
+
+def test_side_staking(
+    web3,
+    config,
+    publisher_wallet,
+    consumer_wallet,
+    another_consumer_wallet,
+    factory_deployer_wallet,
+    factory_router,
+):
+    swap_fee = int(1e15)
+    swap_market_fee = int(1e15)
+    vested_blocks = 2500000
+    initial_ocean_liquidity = web3.toWei("10", "ether")
+
+    ocean_token = ERC20Token(web3, get_address_of_type(config, "Ocean"))
+
+    # Deploy erc721 and erc20 data token
+    erc721, erc20 = deploy_erc721_erc20(
+        web3, config, consumer_wallet, consumer_wallet, web3.toWei("10000", "ether")
+    )
+
+    if ocean_token.balanceOf(consumer_wallet.address) == 0:
```

Comment:
can we move these in some helpers? I get the feeling this is a common case: if balance falls, send some OCEAN.